src/Response/GetPickupPointsResponse.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Answear\BoxNowBundle\Response;
+
+use Answear\BoxNowBundle\DTO\PickupPointDTO;
+use Webmozart\Assert\Assert;
+
+class GetPickupPointsResponse implements ResponseInterface
+{
+    /** @var PickupPointDTO[] */
+    private array $pickupPoints = [];
+
+    public function __construct(private readonly array $data)
+    {
+        $this->handleData();
+    }
+
+    public function handleData(): void
+    {
+        $this->validatePickupPoints($this->data['data'] ?? []);
+
+        foreach ($this->data['data'] as $pickupPoint) {
+            $this->pickupPoints[] = new PickupPointDTO(
+                id: $pickupPoint['id'],
+                type: $pickupPoint['type'],
+                name: $pickupPoint['name'],
+                address: $pickupPoint['addressLine1'],
+                title: $pickupPoint['title'] ?? null,
+                image: $pickupPoint['image'] ?? null,
+                latitude: isset($pickupPoint['lat']) ? (float) $pickupPoint['lat'] : null,
+                longitude: isset($pickupPoint['lng']) ? (float) $pickupPoint['lng'] : null,
+                postalCode: $pickupPoint['postalCode'] ?? null,
+                country: $pickupPoint['country'] ?? null,
+                note: $pickupPoint['note'] ?? null,
+                additionalAddress: $pickupPoint['addressLine2'] ?? null,
+                expectedDeliveryTime: $pickupPoint['expectedDeliveryTime'] ?? null,
+                region: $pickupPoint['region'] ?? null,
+            );
```

Comment:
As in the previous discussion. I do it because of the readability of the code and ease of future maintenance.

For example, if there is a new parameter that is not null, there will be no misalignment of parameters. What do you think about it in general?